execute/plugin.go

```diff
@@ -165,166 +163,6 @@ func getPendingExecutedReports(
 	return groupedCommits, nil
 }
 
-// Observation collects data across two phases which happen in separate rounds.
-// These phases happen continuously so that except for the first round, every
-// subsequent round can have a new execution report.
-//
-// Phase 1: Gather commit reports from the destination chain and determine
-// which messages are required to build a valid execution report.
-//
-// Phase 2: Gather messages from the source chains and build the execution
-// report.
-// nolint:gocyclo // todo
-func (p *Plugin) Observation(
-	ctx context.Context, outctx ocr3types.OutcomeContext, _ types.Query,
-) (types.Observation, error) {
-	var err error
-	var previousOutcome exectypes.Outcome
-
-	if outctx.PreviousOutcome != nil {
-		previousOutcome, err = exectypes.DecodeOutcome(outctx.PreviousOutcome)
-		if err != nil {
-			return types.Observation{}, fmt.Errorf("unable to decode previous outcome: %w", err)
-		}
-		p.lggr.Infow("decoded previous outcome", "previousOutcome", previousOutcome)
-	}
```

Comment:
@dimkouv @makramkd we have to allow nil outcomes because the previous outcome might be nil. May as well embrace it instead of turning it into an edge case.